controllers/subscriptions.go

```diff
@@ -153,3 +155,71 @@ func (r *StorageSystemReconciler) isVendorCsvReady(instance *odfv1alpha1.Storage
 		return err
 	}
 }
+
+// GetSubscriptions returns all required Subscriptions for the given StorageKind
+func GetSubscriptions(k odfv1alpha1.StorageKind) []*operatorv1alpha1.Subscription {
+
+	subscriptions := []*operatorv1alpha1.Subscription{}
+	if k == StorageClusterKind {
+		subscriptions = GetStorageClusterSubscriptions()
+	} else if k == FlashSystemKind {
+		subscriptions = GetFlashSystemClusterSubscriptions()
+	}
+
+	return subscriptions
+}
+
+// GetStorageClusterSubscription return subscription for StorageCluster
+func GetStorageClusterSubscriptions() []*operatorv1alpha1.Subscription {
+	noobaaSubscription := &operatorv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      NoobaaSubscriptionName,
+			Namespace: OperatorNamespace,
+		},
+		Spec: &operatorv1alpha1.SubscriptionSpec{
+			CatalogSource:          NoobaaSubscriptionCatalogSource,
+			CatalogSourceNamespace: NoobaaSubscriptionCatalogSourceNamespace,
+			Package:                NoobaaSubscriptionPackage,
+			Channel:                NoobaaSubscriptionChannel,
+			StartingCSV:            NoobaaSubscriptionStartingCSV,
+			InstallPlanApproval:    operatorv1alpha1.ApprovalAutomatic,
+		},
+	}
+
+	ocsSubscription := &operatorv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OcsSubscriptionName,
+			Namespace: OperatorNamespace,
+		},
+		Spec: &operatorv1alpha1.SubscriptionSpec{
+			CatalogSource:          OcsSubscriptionCatalogSource,
+			CatalogSourceNamespace: OcsSubscriptionCatalogSourceNamespace,
+			Package:                OcsSubscriptionPackage,
+			Channel:                OcsSubscriptionChannel,
+			StartingCSV:            OcsSubscriptionStartingCSV,
+			InstallPlanApproval:    operatorv1alpha1.ApprovalAutomatic,
+		},
+	}
+
+	return []*operatorv1alpha1.Subscription{noobaaSubscription, ocsSubscription}
+}
+
+// GetFlashSystemClusterSubscription return subscription for FlashSystemCluster
+func GetFlashSystemClusterSubscriptions() []*operatorv1alpha1.Subscription {
+	ibmSubscription := &operatorv1alpha1.Subscription{
```

Comment:
What about the IBM CSI subscription, Do we want to handle that also ourselves?

Reply:
If the IBM operator handles that, then we no. I'd rather let them decide how they manage their own components.